.github/workflows/posix.yml

```diff
@@ -46,9 +46,13 @@ jobs:
           - env_file: actions-38.yaml
             pattern: "not slow and not network and not single_cpu"
             extra_apt: "language-pack-zh-hans"
+            # Use the utf8 version as the default, it has no bad side-effect.
             lang: "zh_CN.utf8"
             lc_all: "zh_CN.utf8"
-            name: "Locale: zh_CN.utf8"
+            # Also install zh_CN (its encoding is gb2312) but do not activate it.
+            # It will be temporarily activated during tests with locale.setlocale
+            extra_loc: "zh_CN"
+            name: "Locale: zh_CN.utf8 + extra zh_CN (gb2312)"
```

Comment:
Maybe just `Locale: zn_CN`

Reply:
Ok.

Reply:
Note that the two are totally independent. Adding `extra_loc: "zh_CN"` + the language pack (but not setting the other env vars LC_ALL) in all the "normal" workers could be also a way for the test to execute correctly on all workers. No need to change the default system locale